models/issue.go

```diff
@@ -1568,25 +1569,34 @@ func SearchIssueIDsByKeyword(kw string, repoIDs []int64, limit, start int) (int6
 	return total, ids, nil
 }
 
-func updateIssue(e Engine, issue *Issue) error {
-	_, err := e.ID(issue.ID).AllCols().Update(issue)
-	if err != nil {
-		return err
-	}
-	return nil
+func updateIssueByCols(e Engine, issue *Issue, columns ...string) (err error) {
+	_, err = e.ID(issue.ID).Cols(columns...).Update(issue)
+	return
 }
 
-// UpdateIssue updates all fields of given issue.
-func UpdateIssue(issue *Issue) error {
+// UpdateIssueByAPI updates all allowed fields of given issue.
+func UpdateIssueByAPI(issue *Issue) error {
+	// allowed fields to update
+	columns := []string{"name", "is_closed", "content", "milestone_id",
+		"priority", "num_comments", "ref", "deadline_unix", "created_unix",
+		"updated_unix", "closed_unix", "is_locked"}
+
 	sess := x.NewSession()
 	defer sess.Close()
 	if err := sess.Begin(); err != nil {
 		return err
 	}
-	if err := updateIssue(sess, issue); err != nil {
+	if err := issue.loadPoster(sess); err != nil {
 		return err
 	}
-	if err := issue.loadPoster(sess); err != nil {
+	if !issue.Poster.IsGhost() {
```

Comment:
PosterId should not be changed when edit issue?

Reply:
Don't know ... the function allowed everything until now ... and now we can specify at Line 1580++